FilteringAllocationIT testTransientSettingsStillApplied failure

```
java.lang.AssertionError: shard on node_t0 but should only be on the include node list: node_t5,node_t4,node_t3
	at __randomizedtesting.SeedInfo.seed([3B129C65D01A4727:A7A99B21FD38F20A]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.elasticsearch.cluster.allocation.FilteringAllocationIT.testTransientSettingsStillApplied(FilteringAllocationIT.java:213)
``` 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-unix-compatibility/os=debian-8&&immutable/171/console
https://gradle.com/s/mjyiygpvwyuja

Does not reproduce 
```
./gradlew :server:integTest --tests "org.elasticsearch.cluster.allocation.FilteringAllocationIT.testTransientSettingsStillApplied" \
  -Dtests.seed=3B129C65D01A4727 \
  -Dtests.security.manager=true \
  -Dtests.locale=ar \
  -Dtests.timezone=Indian/Cocos \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

--- a/server/src/test/java/org/elasticsearch/cluster/allocation/FilteringAllocationIT.java
+++ b/server/src/test/java/org/elasticsearch/cluster/allocation/FilteringAllocationIT.java
@@ -181,7 +181,6 @@ public class FilteringAllocationIT extends ESIntegTestCase {
         assertEquals("invalid IP address [192.168.1.1.] for [" + filterSetting.getKey() + ipKey + "]", e.getMessage());
     }
 
-    @AwaitsFix( bugUrl = "https://github.com/elastic/elasticsearch/issues/45003")
     public void testTransientSettingsStillApplied() {
         List<String> nodes = internalCluster().startNodes(6);
         Set<String> excludeNodes = new HashSet<>(nodes.subList(0, 3));

--- a/test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java
@@ -900,7 +900,7 @@ public abstract class ESIntegTestCase extends ESTestCase {
      * using the cluster health API.
      */
     public ClusterHealthStatus waitForRelocation(ClusterHealthStatus status) {
-        ClusterHealthRequest request = Requests.clusterHealthRequest().waitForNoRelocatingShards(true);
+        ClusterHealthRequest request = Requests.clusterHealthRequest().waitForNoRelocatingShards(true).waitForEvents(Priority.LANGUID);
         if (status != null) {
             request.waitForStatus(status);
         }

